metadata_layer.py

```diff
@@ -1183,11 +1193,9 @@ def apply(self, _db, collection, replace=None):
             if link.rel in Hyperlink.CIRCULATION_ALLOWED:
                 link_obj = link_objects[link]
                 if replace.mirror:
-                    for mirror_type in replace.mirror.keys():
-                        if replace.mirror[mirror_type]:
-                            # We need to mirror this resource. If it's an image, a
-                            # thumbnail may be provided as a side effect.
-                            self.mirror_link(pool, data_source, link, link_obj, replace, mirror_type)
+                    mirror_type = replace.mirror_type_for(link)
+                    # We need to mirror this resource.
+                    self.mirror_link(pool, data_source, link, link_obj, replace, mirror_type)
```

Comment:
However you resolve the `mirror_type_for` issue, it's now possible to put _all_ of this logic inside `mirror_link`. You're already passing the LinkData and the ReplacementPolicy into that method, and that gives it all the information necessary to figure out whether, how and where to do the mirroring.